components/timezone/src/windows_tz.rs

```diff
@@ -0,0 +1,200 @@
+    pub fn as_borrowed(&self) -> WindowsTimeZoneMapperBorrowed {
+        WindowsTimeZoneMapperBorrowed {
+            data: self.data.get(),
+        }
+    }
+}
+
+/// A borrowed wrapper around the windows time zone mapper, returned by
+/// [`WindowsTimeZoneMapper::as_borrowed()`].
+#[derive(Debug)]
+pub struct WindowsTimeZoneMapperBorrowed<'a> {
+    data: &'a WindowsZonesToBcp47MapV1<'a>,
+}
+
+impl<'a> WindowsTimeZoneMapperBorrowed<'a> {
+    /// Returns the BCP-47 id for a provided windows time zone string with a case sensitive query.
+    ///
+    /// This method will return the canonical identifier for an area as no
+    /// territory/geo name was provided, so the mapper will use default the default M49 World code ("001").
+    ///
+    /// ```rust
+    /// use icu_timezone::{WindowsTimeZoneMapper, TimeZoneBcp47Id};
+    /// use tinystr::tinystr;
+    ///
+    /// let windows_tz_mapper_owned = WindowsTimeZoneMapper::new();
+    /// let windows_tz_mapper = windows_tz_mapper_owned.as_borrowed();
+    ///
+    /// let bcp47_id = windows_tz_mapper.windows_tz_to_bcp47_id("Central Standard Time");
+    /// assert_eq!(bcp47_id, Some(TimeZoneBcp47Id(tinystr!(8, "uschi"))));
+    /// ```
+    pub fn windows_tz_to_bcp47_id(&self, windows_tz: &str) -> Option<TimeZoneBcp47Id> {
+        self.windows_tz_to_bcp47_id_with_region(windows_tz, None)
+    }
+
+    /// Returns the IANA identifier(s) for a provided windows zone bytes and an
+    /// optional geo_name bytes representing a [`WindowsRegion`]. If not provided,
+    /// geo_name will be set to the default of `WindowsRegion`.
+    ///
+    /// If a `WindowsRegion` is provided, then the returned is not guaranteed to
+    /// be a single identifier. The value may be a space delimited list of IANA
+    /// identifiers for the designated region.
+    ///
+    /// ```rust
+    /// use icu_timezone::{WindowsTimeZoneMapper, TimeZoneBcp47Id};
+    /// use icu_provider::prelude::icu_locale_core::subtags::Region;
+    /// use tinystr::tinystr;
+    ///
+    /// let win_tz_mapper_owned = WindowsTimeZoneMapper::new();
+    /// let win_tz_mapper = win_tz_mapper_owned.as_borrowed();
+    ///
+    /// let region = None;
+    /// let bcp47_id = win_tz_mapper.windows_tz_to_bcp47_id_with_region("Central Standard Time", region);
+    /// assert_eq!(bcp47_id, Some(TimeZoneBcp47Id(tinystr!(8, "uschi"))));
+    ///  
+    /// let region = Some(Region::try_from_str("US").unwrap());
+    /// let bcp47_id = win_tz_mapper.windows_tz_to_bcp47_id_with_region("Central Standard Time", region);
+    /// assert_eq!(bcp47_id, Some(TimeZoneBcp47Id(tinystr!(8, "uschi"))));
+    ///  
+    /// let region = Some(Region::try_from_str("CA").unwrap());
+    /// let bcp47_id = win_tz_mapper.windows_tz_to_bcp47_id_with_region("Central Standard Time", region);
+    /// assert_eq!(bcp47_id, Some(TimeZoneBcp47Id(tinystr!(8, "cawnp"))));
+    ///   
+    /// let region = Some(Region::try_from_str("ZZ").unwrap());
```

Comment:
This time zone is actually a weird one and probably shouldn't be in the docs/doctests?

It looks like this is actually missing in cldr's [windowsTimezone](https://github.com/unicode-org/cldr/blob/main/common/supplemental/windowsZones.xml#L96) data, but it is present in cldr-json's [windowsZones](https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/windowsZones.json#L696) data.

I think I've seen bcp-47 file that this zone is deprecated, but I'm not entirely sure where the misalignment comes from.

Reply:
In the latest TZDB release, CST6CDT is now an alias (link) to America/Chicago. CLDR may or may not have updated to that?

Reply:
Looks like the deprecated zones were removed a [couple weeks ago](https://github.com/unicode-org/cldr/commit/ac0de544c40ab1e9916f8a34f81bf2840f04583b), so I was probably using an outdated version of CLDR locally then.